qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -131,32 +137,39 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
-        y = self._encode_labels(y)
+        X, y = self._validate_input(X, y)
         return ClassifierMixin.score(self, X, y, sample_weight)
 
-    def _fit_and_encode_labels(self, y: np.ndarray) -> np.ndarray:
-        """Fits label or one-hot encoder and converts categorical target data."""
-
-        if isinstance(y[0], str):
-            # string data is assumed to be categorical
-
-            # OneHotEncoder expects data with shape (n_samples, n_features) but
-            # LabelEncoder expects shape (n_samples,) so set desired shape
-            y = y.reshape(-1, 1) if self._one_hot else y
-            self._target_encoder.fit(y)
-            y = self._target_encoder.transform(y)
-
-        return y
-
-    def _encode_labels(self, y: np.ndarray) -> np.ndarray:
-        """Converts categorical target data using label or one-hot encoding."""
-
-        if isinstance(y[0], str):
-            # string data is assumed to be categorical
+    def _validate_input(self, X: np.ndarray, y: np.ndarray = None) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
Given how you word the start of docstring would this be better named `_prepare_input`? It might lead to less confusion to someone reading the code, where its used, as it seems to be saying it validates input which in my mind would be to check format etc to validate it conforms.

Reply:
I prefer to keep `_validate_input` as the similar method is already in `NeuralNetwork` and change the docstring. Updated the docstring.